packages/navie/src/lib/get-most-recent-messages.ts

```diff
@@ -0,0 +1,35 @@
+import { PromptInteractionEvent } from '../interaction-history';
+import type Message from '../message';
+
+const DEFAULT_HISTORICAL_MESSAGES = (() => {
+  const env = process.env.APPMAP_NAVIE_HISTORICAL_MESSAGES;
+  if (env) {
+    const value = parseInt(env, 10);
+    if (Number.isInteger(value) && value > 0) {
+      return value;
+    }
+  }
+
+  // Default one user message, one assistant message
+  return 2;
+})();
```

Comment:
We could consider bumping this number a bit, but ultimately the SOTA approach looks to be an intermediate RAG step over the conversation history.